dev/benchmarks/macrobenchmarks/lib/src/web/bench_material_3_semantics.dart

```diff
@@ -0,0 +1,147 @@
+      final AggregatedTimedBlock compileChildrenBlock = timings.getAggregated('Semantics.compileChildren');
+      profile!.addTimedBlock(semanticsBlock, reported: true);
+      profile!.addTimedBlock(getFragmentBlock, reported: true);
+      profile!.addTimedBlock(compileChildrenBlock, reported: true);
+    }
+
+    super.frameDidDraw();
+    FlutterTimeline.reset();
+  }
+
+  @override
+  Widget createWidget() {
+    return const SingleColumnMaterial3Components();
+  }
+}
+
+/// Measures the cost of semantics when scrolling screens containing Material 3
+/// widgets.
+///
+/// The implementation uses a ListView that jumps the scroll position between
+/// 0 and 1 every frame. Such a small delta is not enough for lazy rendering to
+/// add/remove widgets, but its enough to trigger the framework to recompute
+/// some of the semantics.
+///
+/// The expect output numbers of this benchmarks should be very small as
+/// scrolling a list view should be a matter of shifting some widgets and
+/// updating the projected clip imposed by the viewport. As of June 2023, the
+/// numbers are not great. Semantics consumes >50% of frame time.
+class BenchMaterial3ScrollSemantics extends WidgetRecorder {
+  BenchMaterial3ScrollSemantics() : super(name: benchmarkName);
+
+  static const String benchmarkName = 'bench_material3_scroll_semantics';
+
+  @override
+  Future<void> setUpAll() async {
+    FlutterTimeline.collectionEnabled = true;
+    super.setUpAll();
+    SemanticsBinding.instance.ensureSemantics();
+  }
+
+  @override
+  Future<void> tearDownAll() async {
+    FlutterTimeline.reset();
+  }
+
+  @override
+  void frameDidDraw() {
+    final AggregatedTimings timings = FlutterTimeline.collect();
+    final AggregatedTimedBlock semanticsBlock = timings.getAggregated('SEMANTICS');
+    final AggregatedTimedBlock getFragmentBlock = timings.getAggregated('Semantics.GetFragment');
+    final AggregatedTimedBlock compileChildrenBlock = timings.getAggregated('Semantics.compileChildren');
+    profile!.addTimedBlock(semanticsBlock, reported: true);
+    profile!.addTimedBlock(getFragmentBlock, reported: true);
+    profile!.addTimedBlock(compileChildrenBlock, reported: true);
+
+    super.frameDidDraw();
+    FlutterTimeline.reset();
+  }
+
+  @override
+  Widget createWidget() => _ScrollTest();
+}
+
+class _ScrollTest extends StatefulWidget {
+  @override
+  State<_ScrollTest> createState() => _ScrollTestState();
+}
+
+class _ScrollTestState extends State<_ScrollTest> with SingleTickerProviderStateMixin {
+  late final Ticker ticker;
+  late final ScrollController scrollController;
+
+  @override
+  void initState() {
+    super.initState();
+
+    scrollController = ScrollController();
+
+    bool forward = true;
+
+    // A one-off timer is necessary to allow the framework to measure the
+    // available scroll extends before the scroll controller can be exercised
+    // to change the scroll position.
+    Timer.run(() {
```

Comment:
Should we check for `mounted`?

Reply:
It's already mounted by the time `initState` is called.

Reply:
I mean inside `Timer.run(() { ...here... });`

Reply:
Oh, I don't think we need it. `mounted` being false inside the timer implies the benchmark ended before pumping any interesting frames we care to measure, which can't happen because `WidgetRecorder` waits for a minimum number of frames before stopping the benchmark.